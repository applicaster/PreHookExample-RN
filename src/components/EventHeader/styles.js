import {
  Dimensions,
  Platform,
  StyleSheet,
} from 'react-native';

const iOS = 'ios';
const width = Dimensions.get('window').width;
export const styles = StyleSheet.create({
  eventHeader: {
    flexDirection: 'row',
    paddingHorizontal: 12,
    paddingTop: 12,
  },
  eventHeaderOverlay: {
    backgroundColor: 'transparent',
    position: 'absolute',
    top: 0,
    left: 0,
    width: width - 20,
    zIndex: 2,
  },
  eventHeaderInfo: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    paddingLeft: 10,
    height: 43,
    width: 220,
  },
  eventHeaderInfoWithoutUserName: {
    justifyContent: 'center',
  },
  avatar: {
    height: 40,
    width: 40,
  },
  name: {
    fontSize: 14,
    marginTop: (Platform.OS === iOS) ? -3 : -4,
    marginBottom: -2,
    fontWeight: '600',
  },
  userName: {
    fontSize: 10,
  },
  socialIconContainer: {
    right: 9,
    top: 9,
    position: 'absolute',
  },
});
