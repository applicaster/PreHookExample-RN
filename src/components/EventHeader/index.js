import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  Text,
  View,
} from 'react-native';
import EventTimestamp from '../EventTimestamp';
import SocialIcon from '../SocialIcon';
import { styles } from './styles';

class EventHeader extends Component {
  renderSocialIcon() {
    const { overlay: iconOverImage } = this.props;
    const rightOffsetStyle = { right: (iconOverImage) ? 18 : 9 };
    const tintColorStyle = { tintColor: (iconOverImage) ? this.context.secondaryColor : this.context.textColor };
    
    return (
      <View style={[styles.socialIconContainer, rightOffsetStyle]}>
        <SocialIcon socialNetwork={this.props.source} tintColorStyle={tintColorStyle} />
      </View>);
  }

  renderUserName() {
    const { userName, overlay: textOverImage } = this.props;
    const colorStyle = { color: (textOverImage) ? this.context.secondaryColor : this.context.textColor };
    
    return (userName) ? <Text style={[styles.userName, colorStyle]}>@{userName}</Text> : null;
  }

  renderName() {
    const { name, overlay: textOverImage } = this.props;
    const colorStyle = { color: (textOverImage) ? this.context.secondaryTextColor : this.context.mainColor };

    return <Text style={[styles.name, colorStyle]}>{name}</Text>;
  }

  renderTimestamp() {
    const { createdAt, overlay: textOverImage } = this.props;
    const colorStyle = { color: (textOverImage) ? this.context.secondaryColor : this.context.textColor };

    return <EventTimestamp timestamp={createdAt} colorStyle={colorStyle} />;
  }

  renderHeaderInfo() {
    const { userName } = this.props;
    return (
      <View style={[styles.eventHeaderInfo, !userName && styles.eventHeaderInfoWithoutUserName]}>
        <View style={styles.userInfoContainer}>
          {this.renderName()}
          {this.renderUserName()}
        </View>
        <View style={styles.timestampContainer}>
          {this.renderTimestamp()}
        </View>
      </View>
    );
  }

  renderAvatarImage() {
    const { avatarImageUrl } = this.props;
    return <Image style={styles.avatar} source={{ uri: avatarImageUrl }} />;
  }

  render() {
    const { overlay } = this.props;
    return (
      <View style={[styles.eventHeader, overlay && styles.eventHeaderOverlay]}>
        {this.renderAvatarImage()}
        {this.renderHeaderInfo()}
        {this.renderSocialIcon()}
      </View>
    );
  }
}

EventHeader.propTypes = {
  avatarImageUrl: PropTypes.string,
  createdAt: PropTypes.number,
  name: PropTypes.string,
  overlay: PropTypes.bool,
  source: PropTypes.string,
  userName: PropTypes.string,
};

EventHeader.contextTypes = {
  mainColor: PropTypes.string,
  secondaryColor: PropTypes.string,
  textColor: PropTypes.string,
  secondaryTextColor: PropTypes.string,
};

export default EventHeader;
