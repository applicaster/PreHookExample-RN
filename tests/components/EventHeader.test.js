import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventHeader from '../../src/components/EventHeader';
import { event } from './eventTestDataUtils';

describe('EventHeader', () => {
  test('component renders correctly', () => {
    const someEvent = event(1);
    const html = renderer.create(
      <EventHeader
        avatarImageUrl={someEvent.user.avatarImageUrl}
        createdAt={someEvent.createdAt}
        name={someEvent.name}
        overlay={false}
        type={someEvent.type}
        userName={someEvent.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when overlayed', () => {
    const someEvent = event(1);
    const html = renderer.create(
      <EventHeader
        avatarImageUrl={someEvent.user.avatarImageUrl}
        createdAt={someEvent.createdAt}
        name={someEvent.name}
        overlay
        type={someEvent.type}
        userName={someEvent.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for Instagram event', () => {
    const someEvent = event(1, 'image', 'instagram', 'applicaster');
    const html = renderer.create(
      <EventHeader
        avatarImageUrl={someEvent.user.avatarImageUrl}
        createdAt={someEvent.createdAt}
        name={someEvent.name}
        overlay
        type={someEvent.type}
        userName={someEvent.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for Twitter event', () => {
    const someEvent = event(1, 'image', 'twitter', 'applicaster');
    const html = renderer.create(
      <EventHeader
        avatarImageUrl={someEvent.user.avatarImageUrl}
        createdAt={someEvent.createdAt}
        name={someEvent.name}
        overlay
        type={someEvent.type}
        userName={someEvent.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for Facebook event', () => {
    const someEvent = event(1, 'image', 'facebook', null);
    const html = renderer.create(
      <EventHeader
        avatarImageUrl={someEvent.user.avatarImageUrl}
        createdAt={someEvent.createdAt}
        name={someEvent.name}
        overlay
        type={someEvent.type}
        userName={someEvent.user.userName}
      />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
