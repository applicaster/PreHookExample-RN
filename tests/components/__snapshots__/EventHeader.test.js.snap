// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHeader component renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 12,
      "paddingTop": 12,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "url",
      }
    }
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#7ED321",
          "fontSize": 14,
          "marginBottom": -2,
        }
      }
    >
      Carlos Pinto
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#DEDEDE",
          "fontSize": 10,
        }
      }
    >
      @
      cpinto7
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "bottom": 0,
          "color": "#FFFFFF",
          "fontSize": 12,
          "left": 10,
          "position": "absolute",
        }
      }
    >
      Invalid date
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 30,
        },
        Object {
          "position": "absolute",
          "right": 9,
          "top": 9,
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï…­
  </Text>
</View>
`;
